src/python/pants/conftest.py

```diff
@@ -42,3 +44,35 @@ def pytest_sessionfinish(session) -> None:
     # Technically unecessary, but nice if people are running tests directly from repo
     # (not using pants).
     namespace_init_path.unlink()
+
+
+@pytest.fixture(autouse=True, scope="session")
```

Comment:
This code should be fast enough, but it is worth noting we're being bit here by an issue our users have often brought up: scope is effectively impotent in the face of the way we run tests.

Reply:
Oh yeah, I just didn't want it firing for every test case when we could do better 🤷‍♂️

Reply:
Also scope isn't impotent. _Session_ scope is though. The other scopes are legitimate

Reply:
Would an `__init_subclass__` production runtime check be "better"? That would also loop in 3rdparty plugin authors.

Reply:
I thought about it, but decided to scope this to just our repo for now. I'd honestly approve that PR though. 
We should be intentional though, so this is a good first step.